.github/workflows/build.yml

```diff
@@ -296,6 +296,7 @@ jobs:
         run: |
           cargo check --release
           cargo check --no-default-features --features=no-std --release
+          cargo check --no-default-features --features=futures --release
```

Comment:
I've always been a bit confused by our use of features `std` and `no-std`. Why wouldn't the following check fail?

https://github.com/lightningdevkit/rust-lightning/blob/7bc52aa62cb3c32f51b242ad3fa0ae16aee80cec/lightning/src/lib.rs#L60-L61

Reply:
> In our other no-std CI checks we do it in specific crates, rather than the root workspace.

Yes, but there we don't do checks for `futures`? Or am I misunderstanding what you're saying?

Reply:
Ah, then I really missed the point before. Now testing with futures in https://github.com/lightningdevkit/rust-lightning/pull/1744/commits/c30662af606476d615650ffc44e050c275b77a5c.

Reply:
Alright, after CI wasn't happy, I now moved the `futures` test to its own step and enabled it for a few toolchains/platforms in 2966a78.